src/main/java/me/jellysquid/mods/lithium/mixin/gen/biome_noise_cache/BiomeLayerSamplerMixin.java

```diff
@@ -28,8 +36,26 @@ private void init(LayerFactory<CachingLayerSampler> factory, CallbackInfo ci) {
      * @author gegy1000
      */
     @Overwrite
-    public Biome sample(int x, int y) {
+    public Biome sample(Registry<Biome> registry, int i, int j) {
+        // [VanillaCopy]
         CachingLayerSampler tlSampler = this.tlSampler.get();
-        return this.getBiome(tlSampler.sample(x, y));
+
+        int k = tlSampler.sample(i, j);
+        RegistryKey<Biome> registryKey = Biomes.fromRawId(k);
+        if (registryKey == null) {
+            throw new IllegalStateException("Unknown biome id emitted by layers: " + k);
+        } else {
+            Biome biome = (Biome)registry.get(registryKey);
+            if (biome == null) {
+                if (SharedConstants.isDevelopment) {
+                    throw (IllegalStateException) Util.throwOrPause(new IllegalStateException("Unknown biome id: " + k));
+                } else {
+                    LOGGER.warn("Unknown biome id: ", k);
+                    return (Biome)registry.get(Biomes.fromRawId(0));
```

Comment:
These casts are unnecessary (they come from the decompiler), and this logger message is malformed (a Vanilla bug?).

```suggestion
            Biome biome = registry.get(registryKey);
            if (biome == null) {
                if (SharedConstants.isDevelopment) {
                    throw Util.throwOrPause(new IllegalStateException("Unknown biome id: " + k));
                } else {
                    LOGGER.warn("Unknown biome id: {}", k);
                    return registry.get(Biomes.fromRawId(0));
```

Reply:
To be honest this doesn't even need to be an overwrite, but it was originally.

Reply:
Might as well commit your suggestions. I'll leave the decision over whether or not to convert this to a redirect up to the project maintainers.